airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/migrate_to_poetry/pipeline.py

```diff
@@ -0,0 +1,475 @@
+
+
+class CheckIsMigrationCandidate(Step):
+
+    context: ConnectorContext
+
+    title = "Check if the connector is a candidate for migration to poetry."
+    airbyte_repo = git.Repo(search_parent_directories=True)
+
+    async def _run(self) -> StepResult:
+        connector_dir_entries = await (await self.context.get_connector_dir()).entries()
+        if self.context.connector.language not in [ConnectorLanguage.PYTHON, ConnectorLanguage.LOW_CODE]:
+            return StepResult(
+                step=self,
+                status=StepStatus.SKIPPED,
+                stderr="The connector is not a Python connector.",
+            )
+        if self.context.connector.connector_type != "source":
+            return StepResult(
+                step=self,
+                status=StepStatus.SKIPPED,
+                stderr="The connector is not a source connector.",
+            )
+        if "poetry.lock" in connector_dir_entries and "pyproject.toml" in connector_dir_entries:
+            return StepResult(
+                step=self,
+                status=StepStatus.SKIPPED,
+                stderr="The connector has already been migrated to poetry.",
+            )
+        if "setup.py" not in connector_dir_entries:
+            return StepResult(
+                step=self,
+                status=StepStatus.SKIPPED,
+                stderr="The connector can't be migrated to poetry because it does not have a setup.py file.",
+            )
+        if not self.context.connector.metadata.get("connectorBuildOptions", {}).get("baseImage"):
+            return StepResult(
+                step=self,
+                status=StepStatus.SKIPPED,
+                stderr="The connector can't be migrated to poetry because it does not have a base image defined in the metadata.",
+            )
+
+        if f"{self.context.connector.technical_name}-poetry" in self.airbyte_repo.heads:
+            return StepResult(
+                step=self,
+                status=StepStatus.SKIPPED,
+                stderr="The branch for the migration already exists.",
+            )
+        return StepResult(
+            step=self,
+            status=StepStatus.SUCCESS,
+        )
+
+
+class PoetryInit(Step):
+    context: ConnectorContext
+
+    title = "Generate pyproject.toml and poetry.lock"
+    python_version = "^3.9,<3.12"
+    build_system = {
+        "requires": ["poetry-core>=1.0.0"],
+        "build-backend": "poetry.core.masonry.api",
+    }
+
+    def __init__(self, context: PipelineContext, new_version: str) -> None:
+        super().__init__(context)
+        self.new_version = new_version
+
+    @property
+    def package_name(self) -> str:
+        return self.context.connector.technical_name.replace("-", "_")
+
+    def get_package_info(self, package_info: str) -> dict:
+        package_info_dict = {}
+        for line in package_info.splitlines():
+            if ":" not in line:
```

Comment:
what edge case is this catching? can you add a comment? a log might also be useful in case we do find a package without ":"

Reply:
It's to ignore locally installed packages - I'll add a comment. All these parsing logics are very clumsy, sorry